### PR TITLE
K8s metadata pod

### PR DESCRIPTION
**Pull request by hunchback**
Adding new fields to Pod:
- IP (optional)
- Status
- Node

The mandatory fields are verified via functional test - so if this passes functional we should be ok

### CODE REVIEW

**Comment by hunchback** [comment]
@safchain / @lebauce - kindly review

File: topology/probes/k8s/pod.go [hunk 1]
```diff
@@ -31,7 +31,7 @@ import (
 	"github.com/skydive-project/skydive/probe"
 	"github.com/skydive-project/skydive/topology/graph"
 
-	api "k8s.io/api/core/v1"
```
Comment (nplanel) [inline topology/probes/k8s/pod.go]: using api namespace was much better for readability

File: tests/k8s_test.go [hunk 2]
```diff
@@ -112,24 +112,42 @@ func checkEdgeCreation(t *testing.T, c *CheckContext, from, to *graph.Node, relT
...
-			_, err := checkNodeCreation(t, c, typ, "Name", name)
-			return err
+			obj, err := checkNodeCreation(t, c, typ, "Name", name)
+			if err != nil {
+				return err
+			}
+
+			m := obj.Metadata()
+
+			const fieldDoesNotExist = "DoesNotExist"
```
Comment (nplanel) [inline tests/k8s_test.go]: how a metadata Node that have been queried will reply "DoesNotExist" ?
This should not append, as the metadata (obj.Metadata()[key]) hash key should not exist

File: topology/probes/k8s/pod.go (hunk 1 above)
Comment (hunchback) [reply topology/probes/k8s/pod.go]: but this is not the standard way these packages are accessed from within the k8s code base ... either the default is used such as `v1` or `v1beta1` or if it conflicts with other packages then you append the parent directory such as `corev1` or `metav1beta1`

File: tests/k8s_test.go (hunk 2 above)
Comment (hunchback) [reply tests/k8s_test.go]: ok - removing

File: topology/probes/k8s/pod.go (hunk 1 above)
Comment (nplanel) [reply topology/probes/k8s/pod.go]: so when they will switch (and the probe too) to a v2 of the API you want to rename all the code ?

**Comment by hunchback** [comment]
Its' built this way so one can support several versions within a single
code base, as done here

https://github.com/kubernetes/client-go/blob/master/kubernetes/clientset.go

So when we add v2.Pod support it will be along side v1.Pod (note that each
k8s resource has an independant API versioning)

**Pull request merged**